tools/proofers/codepoint_validator.js

```diff
@@ -149,11 +148,24 @@ $(function () {
                 if(code) {
                     replaceChar = char1 + code;
                 } else {
-                    return;
+                    return match;
                 }
             }
             replaceChar = replaceChar.normalize("NFC");
-            maybeSubstitute();
+            return maybeReplace();
         }
-    });
+
+        text = text.replace(/\[(..)\]/gu, replacer);
```

Comment:
This makes me very wary. So for every 'input' event we will be firing a regex across all of the text. I'm wary of how this will perform on older browsers or slower machines.

Also, this will prevent proofers from inputting some PM-specified diacriticals. For instance, if the PM asked for a certain character be coded as `[:am]` and the user first inserts `[]` and then tries to fill in `:am`, the diacritical will be converted into the a+umlaut before they can insert the `m`. The way it is now there is a single predictable way that the conversion happens (upon `]`) which seems like a boon.

Reply:
It would be interesting to find how much time this really takes. I'll look into it. Perhaps could underclock a machine. 
There was a lot of discussion about the brackets. Perhaps if it hadn't been mentioned that the [ ] key would not work for this purpose no-one would have thought about it. Perhaps we should discuss which option is preferable.